File: legacy/draft/serilalization/scraps/deser_old.py


a_string = "(witch true)(menu (0 spam)(1 eggs))(colour blue)(max_speed 42)"

testa = {}


def deserialize(the_string):
    level = 0
    cur_key = ""
    cur_value = ""
    atKey = True
    cur_data = ""

    for c in the_string:
        print(("    "*level) +c)



        if c == "(":
            level = level + 1
            atKey = True
        elif c == ")":
            level = level - 1

            if level == 0:
                if cur_data:
                    print("d>"+cur_data+"<d")
                    cur_data = ""
                else:
                    print(cur_data)
                    print(">"+cur_key+"< >"+cur_value+"<")
                    cur_key = ""
                    cur_value = ""
            pass
        elif c == " ":
            atKey = False
            pass
        else:
            if level <= 1:
                if atKey:
                    cur_key += c
                else:
                    cur_value += c


    #print(">"+cur_key+"< >"+cur_value+"<")



deserialize(a_string)

File: legacy/draft/serilalization/deser_old.lua

a_string = "(foo bar)"
b_string = "(menu (0 spam)(1 eggs))"
c_string = "(cakeislie true)(menu (0 spam)(1 eggs))(favcolour blue)(answer 42)"

function parseSegment(theString)
    local tmpTable = {}
    local tmpKey = ""
    local tmpVal = ""
    local atKey = true
    local level = 1
    for i = 2, #theString do
        local c = theString:sub(i,i)
        print(c)

        if c == ")" then
            level = level - 1
            if level == 0 then
                --if tmpVal:sub(1,1) == "(" then

                --else
                tmpTable[tmpKey] = tmpVal
                return tmpTable, i
                --end
                --return tmpKey, tmpVal, i
            else
                tmpVal = tmpVal .. c
            end
        elseif c == "(" then
            level = level + 1
            tmpVal = tmpVal .. c
        elseif c == " " then
            if level == 1 then
                atKey = false
            else
                tmpVal = tmpVal .. c
            end

        else
            if atKey then
                tmpKey = tmpKey .. c
            else
                tmpVal = tmpVal .. c
            end
        end
    end
end

function deserialize(theString)
    local f = theString:sub(1,1)
    local curType = ""
    local curData

    if f == "(" then
        curType = "list"
        curData = {}
    else
        curType = "atom"
        curData = ""
    end

    print("parsing "..curType)

    for i = 2, #theString do
        local c = theString:sub(i,i)
        print(c)

        if c == "(" then
            deserialize(theString:sub(i,#theString))
        end
    end

end

deserialize(a_string)

print(parseSegment(b_string))
